autorest/codegen/serializers/sample_serializer.py

```diff
@@ -0,0 +1,160 @@
+            ]
+            for p in params_positional
+        }
+
+        # imports
+        imports = FileImport()
+        namespace = (self.code_model.options["package_name"] or "").replace(
+            "-", "."
+        ) or self.code_model.namespace
+        imports.add_submodule_import(
+            namespace, self.code_model.client.name, ImportType.THIRDPARTY
+        )
+        if credential:
+            imports.add_import("os", ImportType.STDLIB)
+            imports.add_submodule_import(third_party, credential, ImportType.THIRDPARTY)
+
+        return {
+            "imports": FileImportSerializer(imports, True),
+            "client_params": client_params,
+        }
+
+    def serialize_and_write(self) -> None:
+        template = self.env.get_template("sample.py.jinja2")
+        out_path = package_root_folder(
+            self.code_model.namespace, self.namespace_path
+        ) / Path("generated_samples")
+        sample_params = self._prepare_sample_render_param()
+        cls = lambda x: f'"{x}"' if isinstance(x, str) else str(x)
+        failure_info = '"fail to find required param named {%s} in example file {%s}"'
+        # pylint: disable=too-many-nested-blocks
+        for op_group in self.code_model.operation_groups:
+            if self.code_model.options["multiapi"]:
+                api_version_folder = f"{op_group.api_versions[0]}/"
+                sample_params["client_params"]["apiVersion"] = [
+                    "api_version",
+                    f'"{op_group.api_versions[0]}"',
+                ]
+            else:
+                api_version_folder = ""
+            sample_params["operation_group_name"] = (
+                "" if op_group.is_mixin else f".{op_group.property_name}"
+            )
+            for operation in op_group.operations:
+                samples = operation.yaml_data["samples"]
+                if not samples:
+                    continue
+                params_positional = [
+                    p
+                    for p in operation.parameters.positional
+                    if not p.client_default_value
+                ]
+                sample_params["operation_result"] = operation_additional(operation)
+                sample_params["operation_name"] = f".{operation.name}"
+                for key, value in samples.items():
+                    # update client parameters if it is defined in example files
+                    for param in sample_params["client_params"].keys():
+                        p_value = value["parameters"].get(param)
+                        if p_value:
+                            sample_params["client_params"][param][1] = f'"{p_value}"'
+
+                    # prepare method parameters
+                    operation_params = {}
+                    for param in params_positional:
+                        if isinstance(param, BodyParameter):
+                            name = to_lower_camel_case(param.client_name)
```

Comment:
BodyParameter does not have rest_api_name so we have to convert it by ourselves

Reply:
make sense. Move the code here: https://github.com/Azure/autorest.python/pull/1275/files#r909429178